### PR TITLE
EventFilter.error mutes all errors regardless of the cause, #25057

### PR DESCRIPTION
**Pull request by dotta**

### CODE REVIEW

**Comment by ktoso** [comment]
resolves #25057

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
PLS WHITELIST

**Comment by dotta** [comment]
Ah, I just realized I should amend the commit message and change "mutes" with "filter". Shall I make the change and force push or wait for the review process to complete first?

**Review by patriknw — APPROVED** [review]
LGTM, thanks @dotta

**Comment by ktoso** [comment]
Commit message is not a biggie; you can push it now if you want :)

**Review by ktoso — APPROVED** [review]
LGTM thanks! Thanks you made time for fixing it :)

**Comment by dotta** [comment]
> Commit message is not a biggie; you can push it now if you want :)

Done :)

File: akka-testkit/src/test/scala/akka/testkit/TestEventListenerSpec.scala
Comment (raboof) [inline akka-testkit/src/test/scala/akka/testkit/TestEventListenerSpec.scala]: Perhaps also add a testcase that `EventFilter[Error.NoCause]` has the 'old' behavior (only filtering errors that have no cause)?

**Review by raboof — COMMENTED** [review]
Thanks for the contribution! We also need to update the scaladoc on `EventFilter.error()`, other than that LGTM! (one testcase that might be nice to add, but not essential)

File: akka-testkit/src/test/scala/akka/testkit/TestEventListenerSpec.scala
Comment (dotta) [reply akka-testkit/src/test/scala/akka/testkit/TestEventListenerSpec.scala]: If this is the desired behavior, then the current fix should be reworked. I guess an explicit `Error.All` class should be introduced, and `ErrorFilter.error` should use that (as opposed to using `Error.NoCause` as it currently does). This boils down to decide if it's useful for the user to filter only errors that have no cause attached. So, is this what you'd like?

Comment (raboof) [reply akka-testkit/src/test/scala/akka/testkit/TestEventListenerSpec.scala]: Ah, indeed `EventFilter[Error.NoCause.type]` will now filter everything.

I think it would be a useful improvement to allow filtering on errors without a cause, but it's not essential for this PR.

**Review by raboof — APPROVED** [review]

**Review by raboof — CHANGES_REQUESTED** [review]
Oh we should update the scaladoc before merging though

**Comment by dotta** [comment]
> Oh we should update the scaladoc before merging though

@raboof Here is what the `EventFilter.error` scaladoc says:

```
/**
   * Create a filter for Error events which do not have a cause set (i.e. use
   * implicitly supplied Logging.Error.NoCause). See apply() for more details.
   */
```

Which is exactly what the code does. What would you like to change?

**Comment by raboof** [comment]
The `which do not have a cause set` isn't really correct anymore, right? (it's true that this is achieved by passing in Logging.Error.NoCause, but that's more of an implementation detail now, right?)

**Comment by dotta** [comment]
@raboof Agreed. Thanks for convincing me! :) Will make the change and force push

**Comment by dotta** [comment]
@raboof Have another look. I've updated the scaladoc and also added an explicit test that showcase how the `EventFilter[Error.NoCause.type]` works atm (also added a link to the discussion to make it clear the current behavior could be improved).

**Review by raboof — APPROVED** [review]
Looks good, thanks, great contribution!

**Comment by ktoso** [comment]
Great, thanks!

**Pull request merged**
